pkgs/development/libraries/gtk/patches/3.0-hardcode-gsettings.patch

```diff
@@ -0,0 +1,370 @@
+diff --git a/demos/gtk-demo/application.c b/demos/gtk-demo/application.c
+index 289f73a..9ce0466 100644
+--- a/demos/gtk-demo/application.c
++++ b/demos/gtk-demo/application.c
+@@ -384,7 +384,16 @@ demo_application_init (DemoApplication *app)
+   GSettings *settings;
+   GAction *action;
+ 
+-  settings = g_settings_new ("org.gtk.Demo");
++  {
++    g_autoptr(GSettingsSchemaSource) schema_source;
++    g_autoptr(GSettingsSchema) schema;
++    schema_source = g_settings_schema_source_new_from_directory("@gtk@",
++                                                                g_settings_schema_source_get_default(),
++                                                                TRUE, NULL);
++    schema = g_settings_schema_source_lookup(schema_source, "org.gtk.Demo",
```

Comment:
The default schema source is what implements overriding schema with environment variables so when we do not use it, the `extraGSettingsOverrides` NixOS option will be without effect.

We should probably tweak the patch to check the default schema first and only look in the hardcoded one when the schema is missing in the default one.